test/nuxt/composables.test.ts

```diff
@@ -266,6 +266,13 @@ describe('useFetch', () => {
     await useFetch('/api/test', { params: { id: ref('3') } }, '')
     expect.soft(getPayloadEntries()).toBe(baseCount + 3)
   })
+
+  it('should timeout', async () => {
+    const { pending, status, error } = await useFetch('https://httpbin.org/delay/10', { timeout: 1 })
+    await pending.value
```

Comment:
Maye we should await a new promise + setTimeout instead of `await pending.value` here to make it more explicit? 🤔

Reply:
This would probably be a good use case for [vi.waitFor](https://vitest.dev/api/vi.html#vi-waitfor), however it was only introduced in `0.34.5` and Nuxt is pinned to `0.33.0`. Is Nuxt due to upgrade vitest soon?

Reply:
No, it requires https://github.com/danielroe/nuxt-vitest/pull/290 to be merged, which will be a bigger refactor.

Reply:
No problem, I've just awaited a promise as suggested. Do you think 2ms is enough buffer for `useFetch` to settle?